library/components/TNavigationDropdown.vue

```diff
@@ -95,6 +95,19 @@
 				return window.location.pathname.substring(1);
 			},
 		},
+		methods: {
+			async openReport(page) {
+				await this.trackReportOpen(page.title);
+				window.location.href = page.url;
+			},
+			async trackReportOpen(reportTitle) {
+				if (window.ampli) {
+					await window.ampli.reportIsSelected({
+						reportTitle,
+					})
+				}
+			}
+		},
```

Comment:
is there any risk that the ampli function could fail or not complete? Do we need to await it?

Reply:
Hmm... Indeed that is a problem with having the events in topcoat-public and not topcoat-core; things could get out of date. Do we have any method of keeping things up to date that is anything other than manual?

Reply:
Given that it is a page navigation, can we collect the page visit data we already gather when a page is opened and use that? I recall Jeff saying that we have all that information in Amplitude.

I also left a question about this event in the event tracking plan, to clarify what the question is we are answering with this event: https://docs.google.com/presentation/d/1Bv3esoSk-MVA_csGBtS8sb11Z46dwf2zML3O7_y7n_8/edit#slide=id.g13e10e6245e_0_37

Reply:
hi @sareg0 left the comment in docs.

Reply:
If the idea behind this is tracking when a report is opened, we can use this maybe: https://www.docs.developers.amplitude.com/data/sdks/typescript-browser/#page-view-tracking

Reply:
This is more of a dropdown interaction, also I think snyk uses their own page tracking in Ampli and they refused to accept my event on this before.

Reply:
I would like to check this. We should not await for this event, we should check Amplitude docs about events flushing when leaving the page or we should use the `page-view-tracking` instead.

Reply:
` If the Promise is rejected, the await expression throws the rejected value. ` - https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/await


If I understand correctly how [JS handles a throw](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/throw), the execution may be halted unless we deal with any thrown errors, and the navigation could never occur. 


I see that in registry they have written a wrapper that can help with avoiding a slow network request from blocking navigation for too long: https://github.com/snyk/registry/blob/06944be69a2dd4c91fb8a75425cbcfa5162c4e6a/frontend/src/lib/before-navigation-change.ts


If we want to make this navigation tracking more resilient (and we can't use a page visit) then we might want to look into something like the `before-navigation-change` in registry OR do the error handling here and do a 'finally' where we navigate regardless of the promise resolution.

Reply:
Instead of navigation change event, catching the error sounds better to me, what do you think?

Reply:
Also if we use the change event, two problems. It will trigger direct change to browser url too, if I am correct, but this event is just for navigation dropdown on pages.

Reply:
I think that we should not `await` any tracking event. Event tracking agents tends to buffer events and send them in batches when possible.
If we want to ensure events to be sent before a page is re-load or user leaves, we could use something like: https://www.docs.developers.amplitude.com/data/sdks/typescript-browser/migration/#flush-or-onexitpage

Reply:
I think that Amplitude can also be changed to use the Beacon API as the `transport` option: https://developers.amplitude.com/docs/advanced-settings#using-sendbeacon

Reply:
Well in this case, since we should use the 'page exit' approach using, I think it's best to move this event to inside topcoat-core and fire it on page load inside `page_mixing.js` mounted.

We can also move pdf and csv export events to topcoat-core as @sareg0 suggested before inside library_mixin, another benefit of this is that if all the code is inside one project, we can run `ampli status` for health check as well.

Thoughts?

Reply:
So instead of tracking which page user tried to open, just track which page was opened, this will also include initial page load as well.

Reply:
Even using the Beacon API as the `transport` method might help avoid most problems: https://developers.amplitude.com/docs/advanced-settings#using-sendbeacon

https://developer.mozilla.org/en-US/docs/Web/API/Navigator/sendBeacon

Reply:
@sareg0  @egcastro moved tracking functions to topcoat-core mixins and also added `beacon` transport as well, no longer getting any response, which is good.

Also found out that ampli status is important health check command, so all tracking code should be in core, this was also advised by Sara before and I agree.